pkg/controller/logstorage/secrets/secret_controller.go

```diff
@@ -0,0 +1,619 @@
+		return fmt.Errorf("log-storage-controller failed to watch the Secret resource: %w", err)
+	}
+	if err = utils.AddSecretsWatch(c, monitor.PrometheusClientTLSSecretName, truthNS); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch the Secret resource: %w", err)
+	}
+	if err := utils.AddServiceWatch(c, render.ElasticsearchServiceName, render.ElasticsearchNamespace); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch the Service resource: %w", err)
+	}
+	if err := utils.AddServiceWatch(c, esgateway.ServiceName, installNS); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch the Service resource: %w", err)
+	}
+	if err := utils.AddServiceWatch(c, render.LinseedServiceName, installNS); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch the Service resource: %w", err)
+	}
+	return nil
+}
+
+func (r *SecretSubController) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling LogStorage - Secrets")
+
+	// Get LogStorage resource.
+	ls := &operatorv1.LogStorage{}
+	key := utils.DefaultTSEEInstanceKey
+	err := r.client.Get(ctx, key, ls)
+	if err != nil {
+		// Not finding the LogStorage CR is not an error, as a Managed cluster will not have this CR available but
+		// there are still "LogStorage" related items that need to be set up
+		if !errors.IsNotFound(err) {
+			r.status.SetDegraded(operatorv1.ResourceReadError, "An error occurred while querying LogStorage", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+		ls = nil
+		r.status.OnCRNotFound()
+	}
+
+	// We found the LogStorage instance.
+	r.status.OnCRFound()
+
+	// When running in multi-tenant mode, we need to install Linseed in tenant Namespaces. However, the LogStorage
+	// resource is still cluster-scoped (since ES is a cluster-wide resource), so we need to look elsewhere to determine
+	// which tenant namespaces require a Linseed instance. We use the tenant API to determine the set of namespaces that should have a Linseed.
+	var tenant *operatorv1.Tenant
+	if r.multiTenant {
+		if request.Namespace == "" {
+			// In multi-tenant mode, we only handle namespaced reconcile triggers.
+			return reconcile.Result{}, nil
+		}
+
+		// Check if there is a manager in this namespace.
+		tenant, err = utils.GetTenant(ctx, r.client, request.Namespace)
+		if errors.IsNotFound(err) {
+			// No tenant in this namespace. Ignore the update.
+			reqLogger.Info("No Tenant in this Namespace, skip")
+			return reconcile.Result{}, nil
+		} else if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceReadError, "An error occurred while querying Tenant", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Generate keys for Tigera components if we're running in multi-tenant mode and this is a reconcile
```

Comment:
This controller handles both cluster-scoped and tenant-scoped secrets at the moment. This is potentially a little bit confusing, so this might be a good place to even further decompose - e.g., a cluster-scoped secret controller and a tenant-scoped secret controller.